### PR TITLE
Aborted dev deployment status

### PR DESCRIPTION
**Pull request by marekswiecznik**
Issues: MEN-717

I also fixed a bug in `TestDeploymentModelUpdateDeviceDeploymentStatus` (discovered when adding test case with `aborted` status).

@kjaskiewiczz @maciejmrowiec

### CODE REVIEW

File: resources/deployments/deployment_external_test.go [hunk 1]
```diff
@@ -292,6 +294,13 @@ func TestDeploymentGetStatus(t *testing.T) {
 			},
 			OutputStatus: "finished",
 		},
+		"Failed + Aborted": {
+			Stats: map[string]int{
+				DeviceDeploymentStatusFailure:     1,
```
Comment (kjaskiewiczz) [inline resources/deployments/deployment_external_test.go]: it's gofmt ok with this?

Comment (marekswiecznik) [inline resources/deployments/deployment_external_test.go]: nope, corrected, thx

**Comment by kjaskiewiczz** [comment]
+1

**Comment by maciejmrowiec** [comment]
What about update to devices_api docs?
Not sure if I missed but I don't see mechanism that forbids changing aborted status into something else.

**Comment by marekswiecznik** [comment]
Right, I missed that, thx @maciejmrowiec

**Comment by marekswiecznik** [comment]
Added mechanism to forbid setting device deployment status when it is already aborted. @maciejmrowiec

**Pull request merged**